account_invoice_import/wizard/account_invoice_import.py

```diff
@@ -518,13 +536,16 @@ def update_invoice_lines(self, parsed_inv, invoice, seller):
 
     @api.model
     def _prepare_create_invoice_line(self, product, uom, import_line, invoice):
-        ailo = self.env['account.invoice.line']
-        vals = ailo.product_id_change(
-            product.id, uom.id, qty=import_line['qty'], type='in_invoice',
-            partner_id=invoice.partner_id.id,
-            fposition_id=invoice.fiscal_position.id or False,
-            currency_id=invoice.currency_id.id,
-            company_id=invoice.company_id.id)['value']
+        new_line = self.env['account.invoice.line'].new({
+            'invoice_id': invoice,
+            'qty': import_line['qty'],
+            'product_id': product,
+        })
+        new_line._onchange_product_id()
+        vals = {
+            f: new_line._fields[f].convert_to_write(new_line)
```

Comment:
Have made it into a helper function:

def get_values_for_create(instance):
    """ Helper function to get values from object as dict """
    return {
        f: instance._fields[f].convert_to_write(v)
        if not isinstance(instance._fields[f],
            fields._RelationalMulti)
        else v.ids
        for f, v in instance._cache.iteritems()
    }